server/server.go

```diff
@@ -397,11 +397,9 @@ func NewServer(userConfig UserConfig, config Config) (*Server, error) {
 	preWorkflowHooksCommandRunner := &events.DefaultPreWorkflowHooksCommandRunner{
 		VCSClient:             vcsClient,
 		GlobalCfg:             globalCfg,
-		Logger:                logger,
 		WorkingDirLocker:      workingDirLocker,
 		WorkingDir:            workingDir,
-		Drainer:               drainer,
-		PreWorkflowHookRunner: &runtime.PreWorkflowHookRunner{},
+		PreWorkflowHookRunner: runtime.DefaultPreWorkflowHookRunner{},
```

Comment:
Should this be a pointer object?

Reply:
My assumption was that you only need pointers if you have a need to reference the underlying data without copying it but this struct is stateless.